dev/plugins/hu.elte.txtuml.xtxtuml/src/hu/elte/txtuml/xtxtuml/XtxtUML.xtext

```diff
@@ -212,10 +212,18 @@ TUMultiplicity :
 
 // Commons
 
+TUModifiers:
```

Comment:
First of all, using the `&` operator is the proposed solution for this problem in Xtext, despite the lack of detailed error messages (which still can be achieved, as Xtext provides the ability to write custom syntax error messages, see [this stackoverflow question](https://stackoverflow.com/questions/14526524/xtext-customizing-error-msg-by-unordered-groups) or [this Eclipse forum question](https://www.eclipse.org/forums/index.php/t/452330/)).

Secondly, in my opinion, the whole txtUML lacks too many important features to bother with such enhancements that slightly improve the user experience. If we change it to an enum rule, we have to manually write the required checks for duplications and also a code which retrieves the required modifiers from the the enum list.

Not to mention that it is not a clean solution for me to duplicate the visibility modifiers in a `TUModifier` and in a `TUVisibility` rule because at multiple places, only visibility modifiers are used in the language and there is no reason to syntactically enable writing a `static` or an externality keyword there.